src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs

```diff
@@ -260,11 +272,12 @@ private async Task ProcessLogInCallback()
 
     private async Task ProcessLogOut(string returnUrl)
     {
-        if (!await SignOutManager.ValidateSignOutState())
+        if ((Navigation.HistoryEntryState != null && !ValidateSignOutRequestState()) ||
+            // For backcompat purposes, keep SignOutManager working, even though we now use the history.state for this.
+            (Navigation.HistoryEntryState == null && !await SignOutManager.ValidateSignOutState()))
```

Comment:
If we have to keep the `SignOutManager` approach to blocking CSRF anyway, what's the point of having a new mechanism?

Reply:
I am planning to deprecate the SignOutManager in favor of the new way using the NavigationManager, as the SignOutManager relies on local storage, which is not explicitly tied to the navigation action, and requires an additional concept you need to know about.

I am keeping the SignOutManager code around here to avoid the breaking change for people.

Reply:
I have obsoleted the `SignOutManager` as part of this change.